### ISSUE TITLE
prefer-switch: missing some cases

### ISSUE THREAD

**Issue by ikatyang** [palantir/tslint#2685]
### Bug Report

- __TSLint version__: 5.2.0
- __TypeScript version__: 2.3.2
- __Running TSLint via__: CLI & vscode-tslint

#### TypeScript code being linted

```ts
declare const x: any;

if (x === 1 || x === 2 || x === 3) {}
if (x === 1 || x === 2 || x === -3) {}
if (x === 1 || x === 2 || x === null) {}
if (x === 1 || x === 2 || x === true) {}
if (x === 1 || x === 2 || x === false) {}
if (x === 1 || x === 2 || x === `123`) {}
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "prefer-switch": [true]
    }
}
```

#### Actual behavior
```ts
if (x === 1 || x === 2 || x === 3) {} // error message
if (x === 1 || x === 2 || x === -3) {} // no error message
if (x === 1 || x === 2 || x === null) {} // no error message
if (x === 1 || x === 2 || x === true) {} // no error message
if (x === 1 || x === 2 || x === false) {} // no error message
if (x === 1 || x === 2 || x === `123`) {} // no error message
```

#### Expected behavior
```ts
if (x === 1 || x === 2 || x === 3) {} // error message
if (x === 1 || x === 2 || x === -3) {} // error message
if (x === 1 || x === 2 || x === null) {} // error message
if (x === 1 || x === 2 || x === true) {} // error message
if (x === 1 || x === 2 || x === false) {} // error message
if (x === 1 || x === 2 || x === `123`) {} // error message
```
